# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/transportbridge/MediaPipeline.cpp
# Commit: 26913ec8db0b
# Full Hash: 26913ec8db0b350b29145e9502136b29efdeccbc
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Reimplement ssrc/csrc stats cache. r=ng
#   
#   The cache can be a lot simpler than the old one, since libwebrtc handles
#   pruning old entries for us, and since we're now telling libwebrtc when we
#   render frames (so we don't have to make our own stats with guesses of when
# ==============================================================================

diff -r 18d3bc91b71e -r 26913ec8db0b dom/media/webrtc/transportbridge/MediaPipeline.cpp
--- a/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Thu Feb 25 13:07:39 2021 -0600
+++ b/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Mon Mar 15 12:19:57 2021 -0500
@@ -1657,6 +1657,8 @@
   void RenderVideoFrame(const webrtc::VideoFrameBuffer& aBuffer,
                         uint32_t aTimeStamp, int64_t aRenderTime) override {
     mPipeline->mListener->RenderVideoFrame(aBuffer, aTimeStamp, aRenderTime);
+    // This is what drives SSRC/CSRC stats updates in VideoConduit
+    mPipeline->OnFrameDelivered();
   }
 
  private:
@@ -1723,6 +1725,10 @@
   }
 }
 
+void MediaPipelineReceiveVideo::OnFrameDelivered() {
+  (*mConduit->AsVideoSessionConduit())->OnFrameDelivered();
+}
+
 DOMHighResTimeStamp MediaPipeline::GetNow() const { return mConduit->GetNow(); }
 
 DOMHighResTimeStamp MediaPipeline::RtpCSRCStats::GetExpiryFromTime(